### PR TITLE
Core24 237 239 41 finalise entity detail frontend

### PR DESCRIPTION
**Pull request by neb42**
Jira ticket: JIRA-CORE24-237
Jira ticket: JIRA-CORE24-239
Jira ticket: JIRA-CORE24-41

## What have you implemented?

* Validate payloads and responses in api controller so that default values are sent (eg. sending no email addresses gets parsed to an empty array)
* Configure more optional fields
* Change client to use put for updates
* Use disabled form for read only view
* Ensure correct navigation, data flow, alert showing, etc when switching between create/edit/read modes

## Extra/next steps

* The read only mode looks bad. It's functional but not production ready.

### CODE REVIEW

**Review by internetti — CHANGES_REQUESTED** [review]
when clicking save, the 'do you want to leave' alert pops up, that's confusing.

**Review by internetti — CHANGES_REQUESTED** [review]
now I get this error in the console after saving: 
`"Error: No route matches URL "/prm/participants/01HX8ZKRYC2WD73H4H33ZMMVRN""`

also, this might not be from this PR, I don't immediately see something causing this, but when starting to enter something into an input field, there's an error in the console about an uncontrolled input being turned into a controlled input. I had fixed that at some point by setting an empty default value

**Review by internetti — CHANGES_REQUESTED** [review]
and there's a warning about the allowToggle on the accordion, apparently that's not used when allowMultiple is used

**Comment by neb42** [comment]
> and there's a warning about the allowToggle on the accordion, apparently that's not used when allowMultiple is used

This is resolved

File: libs/prm-components/src/lib/hooks/useApiReducer.hook.ts [hunk 1]
```diff
@@ -49,14 +55,12 @@ export const useApiReducer = <T>() => {
             error: action.data,
           };
         case SubmitStatus.IDLE:
-          return { ...state, status: SubmitStatus.IDLE };
+          return defaultState;
```
Comment (internetti) [inline libs/prm-components/src/lib/hooks/useApiReducer.hook.ts:58]: if this is used for the idle case, and not the default case, I'd call it idleState?

**Review by internetti — APPROVED** [review]

**Review by dfigueroa87 — APPROVED** [review]

**Pull request merged**